testsuite/features/step_definitions/common_steps.rb

```diff
@@ -461,6 +461,24 @@
   raise "xpath: #{xpath} not found" unless find(:xpath, xpath).set(true)
 end
 
+When(/I wait until the tree item "([^"]+)" has no sub-list/) do |item|
+  repeat_until_timeout(message: "could still find a sub list for tree item #{item}") do
+    xpath = "//span[contains(text(), '#{item}')]/ancestor::div[contains(@class, 'product-details-wrapper')]/div/i[contains(@class, 'fa-angle-')]"
+    begin
+      find(:xpath, xpath)
+      sleep 1
+    rescue Capybara::ElementNotFound
+      break
+    end
+  end
+end
+
+And(/^I wait until the tree item "([^"]+)" contains "([^"]+)" text$/) do |item, text|
+  within(:xpath, "//span[contains(text(), '#{item}')]/ancestor::div[contains(@class, 'product-details-wrapper')]") do
+    raise "could not find text #{text} for tree item #{item}" unless has_text?(text, wait: DEFAULT_TIMEOUT)
+  end
+end
+
 And(/^I open the sub-list of the product "(.*?)"$/) do |product|
```

Comment:
"When" please

Reply:
that one is from another commit though ;)

Reply:
yeah, but if you could fix it while you're at it ;-)